Automattic/wp-calypso
# A4A > Migrations: Implement Get & Save tagged sites for commissions

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1506

## Proposed Changes

This PR implements:

- API integration to get the tagged sites
- API integration to tag the sites for commissions.
- Some code refactor

## Why are these changes being made?

* To fetch and save the tagged sites for migration

## Testing Instructions

* Open the A4A live link
* Patch D166747-code
* Go to the Migrations: FAQ sections by appending the URL with `/migrations/overview#migration-payment-process-faq`. Click the payment details link(`here` link) and verify that you are redirected to the `/migrations/payment-settings` page. 
* Disable the feature flag by appending the URL with `/migrations?flags=-a4a-tracking-site-migrations#migration-payment-process-faq` and verify you are redirected to `/referrals/payment-settings`

* Go to Migrations: Commissions > Verify that the empty view is shown

<img width="1728" alt="Screenshot 2024-11-21 at 3 44 37 PM" src="https://github.com/user-attachments/assets/1dc94862-a686-4789-8568-1ad08dbcd85e">

* Click the `Tag sites for commission` button > Add a couple of sites > Click the `Add X sites` button > Verify the loading indicator is shown > Wait for the request to be complete 

<img width="821" alt="Screenshot 2024-11-21 at 3 45 33 PM" src="https://github.com/user-attachments/assets/25fb68cf-0efc-44c9-ba9d-3d0f15075a9f">

* Verify that the success notice is shown and the added site is shown in the list as displayed below. 

<img width="1728" alt="Screenshot 2024-11-22 at 3 26 41 PM" src="https://github.com/user-attachments/assets/f86fb72b-eeb2-444f-8fd1-d94a77ff3ce5">

<img width="759" alt="Screenshot 2024-11-21 at 4 28 34 PM" src="https://github.com/user-attachments/assets/a83d7239-ba78-4a3a-9fbc-142faf60ab4e">

<img width="759" alt="Screenshot 2024-11-21 at 4 28 44 PM" src="https://github.com/user-attachments/assets/46e625a8-d9da-4c3c-bdca-c6bec81505f9">


## Pre-merge Checklist
